cirq/study/trial_result.py

```diff
@@ -277,6 +281,12 @@ def __eq__(self, other):
             return NotImplemented
         return self.data.equals(other.data) and self.params == other.params
 
+    def __getitem__(self, key: str) -> np.ndarray:
+        return self._measurements[key]
```

Comment:
Regarding the choice of `measurements` vs `data` (discussed on #1288), I think we should choose the one that is the more common case. My own experience is that the pandas DataFrame is easier to use (because it is more self-documenting), so it would be my preference.

@dabacon @kevinsung @Strilanc WDYT?

Reply:
That would trigger the creation of the data frame.  I am not sure how expensive an option that is, but it would force the creation of the whole ndarray though.  If people are going to do that anyway though, maybe it doesn't matter.

Reply:
We're caching the DataFrame, so it's only going to hit the user once.

Also, DataFrame is well-suited for interactive use (e.g. in ipython, colab etc) while ndarray is a better choice in code where we care about performance. In the latter - since it's generally non-interactive - one can easily write out the longer reference like `result.measurements[key]`, but in interactive use it's nice to be able to write the short `result[key]` and get a DataFrame.